dal/repositories/user_repository.py

```diff
@@ -0,0 +1,34 @@
+from controllers.auth import connect_to_db
+from models.user import User
+
+async def with_db_connection(func):
```

Comment:
you dont need this just init db using beanie once.

Reply:
done

Reply:
no changes been made

Reply:
like this: from models.user import User
from beanie import init_beanie
from typing import Optional